### PR TITLE
More object unit tests

### PR DESCRIPTION
**Pull request by rmarco3e8**
Hello, this is my first pull request so any suggested changes or constructive criticism will be much-appreciated. I tried building off of the existing unit tests for the object class.

I added two test cases. One is a brief test of the setter and getter methods of the script interface of the Object class. The second runs through some tests of a number of methods including setters and getters in the IAPI interface of the Object class. One missing case regarding the get_indexed() and set_indexed() methods is for Vector<StringName> arguments of size greater than 1. I had some trouble getting the methods to update r_valid to true in this case.

### CODE REVIEW

File: tests/test_object.h [hunk 1]
```diff
@@ -87,6 +87,97 @@ TEST_CASE("[Object] Metadata") {
 			meta_list2.size() == 0,
 			"The metadata list should contain 0 items after removing all metadata items.");
 }
+
+TEST_CASE("[Object] IAPI methods") {
+	Object object;
+    bool *rvalid = memnew(bool(false));
```
Comment (Xrayez) [inline tests/test_object.h:93]: I think this adds unnecessary complexity, I'd rewrite this to:
```suggestion
    bool valid = false;
```
and then pass the pointer to methods which use this variable, such as below:
```c++
object.get_indexed(p_names, &valid);
```
Also, this way you don't have to dereference the pointer (`*valid`) whenever you need to reset the `valid` variable for other assertions.

File: tests/test_object.h [hunk 2]
```diff
@@ -87,6 +87,97 @@ TEST_CASE("[Object] Metadata") {
...
+
+    object.set(StringName("script"), "script_source", rvalid);
```
Comment (Xrayez) [inline tests/test_object.h:95]: I'm not 100% sure but I think there's no need to explicitly convert the first argument to `StringName`:

```suggestion
    object.set("script", "script_source", rvalid);
```
The same applies to other assertions.

File: tests/test_object.h [hunk 3]
```diff
@@ -87,6 +87,97 @@ TEST_CASE("[Object] Metadata") {
...
+			*rvalid == true,
+			"get() should update the passed bool to the expected value.");
+    
+    Vector<StringName> p_names;
+    object.set_indexed(p_names, "value", rvalid);
+    CHECK_MESSAGE(
+		*rvalid == false,
+		"set_indexed() should update the passed bool to the expected value for empty Vector<StringName>.");
+
+    *rvalid = true;
```
Comment (Xrayez) [inline tests/test_object.h:114]: Sanity check: should this be reset to `false` (at least for consistency)?

```suggestion
    *rvalid = false;
```

**Comment by akien-mga** [comment]
Sorry for the late review. This was at least partially superseded by #45411. If there are some parts tested here which are not covered in the latest version they might be worth sending as a new PR based on the latest `master` branch (note that the file moved to `tests/core/object/test_object.h`).

**Pull request closed without merging**